homeassistant/components/hive/__init__.py

```diff
@@ -73,9 +123,60 @@ def setup(hass, config):
     session.attributes = Pyhiveapi.Attributes()
     hass.data[DATA_HIVE] = session
 
-    for ha_type, hive_type in DEVICETYPES.items():
-        for key, devices in devicelist.items():
-            if key == hive_type:
-                for hivedevice in devices:
-                    load_platform(hass, ha_type, DOMAIN, hivedevice, config)
+    for ha_type in DEVICETYPES:
+        devicelist = devices.get(DEVICETYPES[ha_type])
+        if devicelist:
+            load_platform(hass, ha_type, DOMAIN, devicelist, config)
+            if ha_type == "climate":
+                hass.services.register(
+                    DOMAIN,
+                    SERVICE_BOOST_HEATING,
+                    heating_boost,
+                    schema=BOOST_HEATING_SCHEMA,
+                )
+            if ha_type == "water_heater":
+                hass.services.register(
+                    DOMAIN,
+                    SERVICE_BOOST_HEATING,
+                    hotwater_boost,
+                    schema=BOOST_HOTWATER_SCHEMA,
+                )
+
     return True
+
+
+def refresh_system(func):
+    """Force update all entities after state change."""
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        func(self, *args, **kwargs)
+        dispatcher_send(self.hass, DOMAIN)
+
+    return wrapper
+
+
+class HiveSession(Entity):
+    """Initiate Hive Session Class."""
+
+    entity_lookup = {}
+    core = None
+    heating = None
+    hotwater = None
+    light = None
+    sensor = None
+    switch = None
+    weather = None
+    attributes = None
+
+    async def async_added_to_hass(self):
+        """When entity is added to Home Assistant."""
+        await super().async_added_to_hass()
+        async_dispatcher_connect(self.hass, DOMAIN, self._update_callback)
+        if self.device_type in SERVICES:
+            self.session.entity_lookup[self.entity_id] = self.node_id
```

Comment:
We need an init method that initializes the instance attributes that we need to access here.